src/client/src/orangehrmClaimPlugin/components/AddExpenseModal.vue

```diff
@@ -0,0 +1,193 @@
+          </oxd-grid-item>
+        </oxd-grid>
+      </oxd-form-row>
+
+      <oxd-form-row>
+        <oxd-grid :cols="2" class="orangehrm-full-width-grid">
+          <oxd-grid-item>
+            <date-input
+              v-model="expense.date"
+              :label="$t('general.date')"
+              :rules="rules.date"
+              :years="yearsArray"
+              required
+            />
+          </oxd-grid-item>
+          <oxd-grid-item>
+            <oxd-input-field
+              v-model="expense.amount"
+              :label="$t('claim.amount')"
+              :rules="rules.amount"
+            />
+          </oxd-grid-item>
+        </oxd-grid>
+      </oxd-form-row>
+
+      <oxd-form-row>
+        <oxd-grid :cols="1" class="orangehrm-full-width-grid">
+          <oxd-grid-item>
+            <oxd-input-field
+              v-model="expense.note"
+              type="textarea"
+              :label="$t('general.note')"
+              :rules="rules.note"
+            />
+          </oxd-grid-item>
+        </oxd-grid>
+      </oxd-form-row>
+
+      <oxd-divider />
+      <oxd-form-actions>
+        <required-text />
+        <oxd-button
+          type="button"
+          display-type="ghost"
+          :label="$t('general.cancel')"
+          @click="onCancel"
+        />
+        <submit-button />
+      </oxd-form-actions>
+    </oxd-form>
+  </oxd-dialog>
+</template>
+
+<script>
+import {APIService} from '@ohrm/core/util/services/api.service';
+import {required, validDateFormat} from '@/core/util/validation/rules';
+import useDateFormat from '@/core/util/composable/useDateFormat';
+import {yearRange} from '@ohrm/core/util/helper/year-range';
+import {OxdDialog} from '@ohrm/oxd';
+import {
+  shouldNotExceedCharLength,
+  digitsOnlyWithDecimalPoint,
+  maxCurrency,
+} from '@ohrm/core/util/validation/rules';
+import ClaimExpenseTypeDropdown from './ClaimExpenseTypeDropdown.vue';
+
+const expenseModel = {
+  expenseType: null,
+  date: null,
+  amount: null,
+  note: null,
+};
+
+export default {
+  name: 'SaveExpense',
+
+  components: {
+    'oxd-dialog': OxdDialog,
+    'claim-expense-type-dropdown': ClaimExpenseTypeDropdown,
+  },
+
+  props: {
+    requestId: {
+      type: Number,
+      required: true,
+    },
+  },
+
+  emits: ['close'],
+
+  setup(props) {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      `api/v2/claim/requests/${props.requestId}/expenses`,
+    );
+
+    const {userDateFormat} = useDateFormat();
+
+    return {
+      http,
+      userDateFormat,
+    };
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      yearsArray: [...yearRange()],
+      expense: {
+        ...expenseModel,
+      },
+      rules: {
+        date: [required, validDateFormat(this.userDateFormat)],
+        note: [shouldNotExceedCharLength(1000)],
+        amount: [
+          required,
+          digitsOnlyWithDecimalPoint,
+          maxCurrency(1000000000),
+          (amount) => {
```

Comment:
feels like a simple regex will be better here. `^\d+\.\d{2}` (please add as a rule and add tests)